From: Feng Kan <fkan@apm.com>
Date: Wed, 4 Dec 2013 13:33:50 -0500
Subject: [PATCH 2/2] arm64: removed using of the mask attribute in the dts for
 reset bit.
Origin: http://lists.infradead.org/pipermail/linux-arm-kernel/2013-December/216350.html

Remove the use of the mask attribute from original reboot driver. The use
of the mask attribute provided unneeded complexity and security risk. Also
fix minor compilation warning.

Signed-off-by: Feng Kan <fkan at apm.com>

[ijc -- obsoleted by upstream syscon-reboot driver reworking, but that is too
 big to backport.]
[bwh: Forward-ported to 3.19. Upstream now has both drivers.]
---
 drivers/power/reset/xgene-reboot.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/power/reset/xgene-reboot.c
+++ b/drivers/power/reset/xgene-reboot.c
@@ -74,8 +74,7 @@ static int xgene_reboot_probe(struct pla
 		return -ENODEV;
 	}
 
-	if (of_property_read_u32(dev->of_node, "mask", &ctx->mask))
-		ctx->mask = 0xFFFFFFFF;
+	ctx->mask = 0x1;
 
 	ctx->dev = dev;
 	ctx->restart_handler.notifier_call = xgene_restart_handler;
